lib/inc/internal/util/agent.hpp

```diff
@@ -0,0 +1,22 @@
+/**
+ * @file
+ * Declares methods for Facter's integration with the puppet-agent package.
+ */
+#pragma once
+
+#include <leatherman/execution/execution.hpp>
+#include <string>
+
+namespace facter { namespace util { namespace agent {
+    inline std::string which(std::string exe) {
```

Comment:
Assuming that this will almost always be called with a string literal, taking the string by value and moving on the return involves only a single construction (as opposed to two when taking the const&)... but that's a micro-optimization, and const& is probably cleaner looking